doc(pubsub): more landing page improvements

coryan opened the pull request:
Some tweaks to the README file and the Doxygen landing pages.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/googleapis/google-cloud-cpp/4978)
<!-- Reviewable:end -->

coryan commented:
/cc: @anguillanneuf

anguillanneuf commented:
Is it okay to fix a few other nits and issues in this PR re the quickstart?

1. `cmake.org][https://cmake.org/download/]` here: https://github.com/googleapis/google-cloud-cpp/blob/master/google/cloud/pubsub/quickstart/README.md
1. This step `./vcpkg install google-cloud-cpp` took 21.24 min on my machine. It will be good to mention that this step could take a long time. 
1. Why does the quickstart require me to have `.build/quickstart [GCP PROJECT] [CLOUD SPANNER INSTANCE] [CLOUD SPANNER DATABASE]`? This wasn't a prerequisite communicated to me earlier. 
1. `.build/quickstart quickstart <project-id> <topid-ic>` returned errors: 

```
E0827 15:53:54.919211000 123145444261888 ssl_utils.cc:565]             load_file: {"created":"@1598568834.919151000","description":"Failed to load file","file":"/Users/tianzi/Projects/vcpkg/buildtrees/grpc/src/097a3f9f89-9c2d39035a.clean/src/core/lib/iomgr/load_file.cc","file_line":72,"filename":"/usr/share/grpc/roots.pem","referenced_errors":[{"created":"@1598568834.919148000","description":"No such file or directory","errno":2,"file":"/Users/tianzi/Projects/vcpkg/buildtrees/grpc/src/097a3f9f89-9c2d39035a.clean/src/core/lib/iomgr/load_file.cc","file_line":45,"os_error":"No such file or directory","syscall":"fopen"}]}
E0827 15:53:54.919746000 123145444261888 ssl_security_connector.cc:413] Could not get default pem root certs.
E0827 15:53:54.919765000 123145444261888 secure_channel_create.cc:132] Failed to create secure subchannel for secure name 'pubsub.googleapis.com'
E0827 15:53:54.919782000 123145444261888 secure_channel_create.cc:50]  Failed to create channel args during subchannel creation.
E0827 15:53:54.919852000 123145444261888 ssl_security_connector.cc:413] Could not get default pem root certs.
E0827 15:53:54.919870000 123145444261888 secure_channel_create.cc:132] Failed to create secure subchannel for secure name 'pubsub.googleapis.com'
E0827 15:53:54.919885000 123145444261888 secure_channel_create.cc:50]  Failed to create channel args during subchannel creation.
Standard exception raised: Empty update
```

coryan commented:
> Is it okay to fix a few other nits in this PR re the quickstart.md?

SGTM.  Thank you for looking at this quickly.

> 
> 1. Use `()` instead of `[]` in `cmake.org][https://cmake.org/download/]`? https://github.com/googleapis/google-cloud-cpp/blob/master/google/cloud/pubsub/quickstart/README.md

Fixed.

> 2. This step `./vcpkg install google-cloud-cpp` took 21.24 min on my machine as a fresh install. It will be good to mention that this step could take a long time.

Done.

> 3. Why does the quickstart require me to have a Spanner instance and database`.build/quickstart [GCP PROJECT] [CLOUD SPANNER INSTANCE] [CLOUD SPANNER DATABASE]`? This wasn't a prerequisite communicated to me earlier.

Fixed, added a section pointing to the online quickstarts.

> 4. ~I tried `.build/quickstart quickstart <project-id> <topid-ic>` next. But it returned errors:~ I missed a step for MacOS users! After running [Platform Specific Notes](https://github.com/googleapis/google-cloud-cpp/blob/master/google/cloud/pubsub/quickstart/README.md#platform-specific-notes), I got `Hello World published with id=1464992386750336`. Looks good!

Thanks.  I moved the warnings about earlier in each section, they were obviously easy to miss.

anguillanneuf commented on google/cloud/pubsub/quickstart/README.md:
```diff
@@ -41,22 +52,25 @@ https://cloud.google.com/docs/authentication/production
    manager such as [chocolatey][choco-cmake-link], and on macOS using
    [homebrew][homebrew-cmake-link]. You can also obtain the software directly
-   from the [cmake.org][https://cmake.org/download/].
+   from the [cmake.org](ttps://cmake.org/download/).
```
nit: (h

coryan replied:
More than a nit, thanks once more.
